ui/src/components/dataset/DatasetNameInput.vue

```diff
@@ -3,12 +3,12 @@
     v-model="datasetNameInput"
     :placeholder="'Dataset name'"
     class="w-full"
-    :messages="'Name for the uploaded dataset'"
+    :messages="props.error ? '' : 'Name of the uploaded dataset'"
     :disabled="props.inputDisabled"
   />
-
-  <div class="va-text-danger text-xs" v-if="props.showDatasetNameError">
-    {{ props.datasetNameError }}
+  <!-- Only one of error and hint are shown at a time -->
```

Comment:
Minor UI change: Showing Input's Hint message as well as Error message is better.